scipy/stats/_continuous_distns.py

```diff
@@ -4219,23 +4220,50 @@ def fit(self, data, *args, **kwds):
 laplace = laplace_gen(name='laplace')
 
 
-def _check_fit_input_parameters(data, args, kwds, fixed_param):
-    if len(args) > 0:
-        raise TypeError("Too many arguments.")
-
-    _remove_optimizer_parameters(kwds)
-
-    if None not in fixed_param:
+def _check_fit_input_parameters(self, data, args, kwds):
+    data = np.asarray(data)
+    floc = kwds.get('floc', None)
+    fscale = kwds.get('fscale', None)
+
+    num_shapes = len(self.shapes) if self.shapes else 0
+    fshape_keys = []
+    fshapes = []
+
+    # user has many options for fixing the shape, so here we standardize it
+    # into 'f' + the number of the shape.
+    if self.shapes:
+        shapes = self.shapes.replace(',', ' ').split()
+        for j, s in enumerate(shapes):
+            key = 'f' + str(j)
+            names = [key, 'f' + s, 'fix_' + s]
+            val = _get_fixed_fit_value(kwds, names)
+            fshape_keys.append(key)
+            fshapes.append(val)
+            if val is not None:
+                kwds[key] = val
+
+    # determine if there are any unknown arguments in kwds
+    keys = [x for x in kwds.keys() if x not in ['loc', 'scale', 'optimizer',
+                                                'floc', 'fscale',
+                                                *fshape_keys]]
+    if keys:
```

Comment:
```suggestion
    known_keys = {'loc', 'scale', 'optimizer', 'floc', 'fscale', *fshape_keys}
    if not known_keys.issuperset(kwds.keys()):
```

Reply:
I added an implementation similar to this – I wanted to keep a copy of the difference in keys to then print out.